cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -92,6 +106,9 @@ class IteratorPointerDereferenceOperator extends Operator, TaintFunction, Iterat
   override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
     input = iteratorInput and
     output.isReturnValue()
+    or
+    input.isReturnValueDeref() and
+    output = getIteratorArgumentOutput(this, 0)
```

Comment:
Can't this be simplified to `output.isParameterDeref(0)`? The charpred should have already established that we're looking at an `operator*` for an iterator.